storage/src/backend/connection.rs

```diff
@@ -449,92 +489,83 @@ impl Connection {
             }
         }
 
-        let current_mirror = self.mirror_state.current.load();
-
-        if let Some(mirror) = current_mirror.as_ref() {
-            // With configuration `auth_through` disabled, we should not intend to send authentication
-            // request to mirror. Mainly because mirrors like P2P/Dragonfly has a poor performance when
-            // relaying non-data requests. But it's still possible that ever returned token is expired.
-            // So mirror might still respond us with status code UNAUTHORIZED, which should be handle
-            // by sending authentication request to the original registry.
-            //
-            // - For non-authentication request with token in request header, handle is as usual requests to registry.
-            //   This request should already take token in header.
-            // - For authentication request
-            //      1. auth_through is disabled(false): directly pass below mirror translations and jump to original registry handler.
-            //      2. auth_through is enabled(true): try to get authenticated from mirror and should also handle status code UNAUTHORIZED.
-            if mirror.config.auth_through
-                || !requesting_auth && headers.contains_key(HEADER_AUTHORIZATION)
-            {
-                let data_cloned = data.as_ref().cloned();
-
-                for (key, value) in mirror.config.headers.iter() {
-                    headers.insert(
-                        HeaderName::from_str(key).unwrap(),
-                        HeaderValue::from_str(value).unwrap(),
-                    );
+        if !self.mirrors.is_empty() {
+            let mut requested_all_mirrors = true;
```

Comment:
Can we rename it to `fallback_due_auth=false` ?

Reply:
Done.